tests/integration/netdev.bats

```diff
@@ -0,0 +1,156 @@
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	ip netns exec "$ns_name" ip link
+	ip netns del "$ns_name"
+	ip link del dev dummy0
+}
+
+@test "move network device to container network namespace and rename" {
+	requires root
+	# create a dummy interface to move to the container
+	ip link add dummy1 type dummy
+	ip link set up dev dummy1
+	ip addr add 169.254.169.14/32 dev dummy1
+
+	update_config ' .linux.netDevices |= { "dummy1": { "name" : "ctr_dummy1" } }
+								| .process.args |= ["ip", "address", "show", "dev", "ctr_dummy1"]'
+	
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	ip netns del "$ns_name"
+	ip link del dev dummy1
+}
+
+@test "move network device to container network namespace and change ipv4 address" {
+	requires root
+	# create a dummy interface to move to the container
+	ip link add dummy1 type dummy
+	ip link set up dev dummy1
+	ip addr add 169.254.169.14/32 dev dummy1
+
+	update_config ' .linux.netDevices |= { "dummy1": { "name" : "ctr_dummy1" , "addresses" : [ "10.0.0.2/24" ]} }
+								| .process.args |= ["ip", "address", "show", "dev", "ctr_dummy1" ]'
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "$output" == *"10.0.0.2/24"* ]]
+
+	ip netns del "$ns_name"
+	ip link del dev dummy1
+}
+
+@test "move network device to container network namespace and change ipv6 address" {
+	requires root
+	# create a dummy interface to move to the container
+	ip link add dummy1 type dummy
+	ip link set up dev dummy1
+	ip addr add 169.254.169.14/32 dev dummy1
+
+	update_config ' .linux.netDevices |= { "dummy1": { "name" : "ctr_dummy1" , "addresses" : [ "10.0.0.2/24" , "2001:db8::2/64" ]} }
+								| .process.args |= ["ip", "address", "show", "dev", "ctr_dummy1" ]'
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "$output" == *"2001:db8::2/64"* ]]
+
+	ip netns del "$ns_name"
+	ip link del dev dummy1
```

Comment:
this guarantees the interface is brought back to the root namespcae, otherwise this command will fail